Adding bazel install rules for gflags

fbudin69500 opened the pull request:
Installing files from an external project generates warning messages (install.bzl).
Warning messages are currently silenced in bazel.rc until install.bzl is updated.

Toward #3129.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/robotlocomotion/drake/6323)
<!-- Reviewable:end -->

jamiesnape commented:
Review status: 0 of 5 files reviewed at latest revision, 7 unresolved discussions.

---

*[tools/bazel.rc, line 12 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6323#-KmWcJTfiyvZyloFOc2a:-KmWcJTfiyvZyloFOc2b:b-vddc2l) ([raw file](https://github.com/robotlocomotion/drake/blob/07ee3d99e3fdc13cce83ac28cc3e720399f484ae/tools/bazel.rc#L12)):*
> ```
> # build --output_filter="^//"
> # Hide warnings from gflags. Warnings are thrown by install.bzl
> # TODO(mwoehlke) Uncomment line above and remove line below once install.bzl
> ```

TODOs should have the author of the TODO's, not the intended resolver. Here, I guess`TODO(fbudin69500, mwoehlke-kitware)` would probably be fine, though.

---

*[tools/BUILD, line 114 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6323#-KmWb-hT76VKs02Y1cTR:-KmWb-hT76VKs02Y1cTS:b-5ajkil) ([raw file](https://github.com/robotlocomotion/drake/blob/07ee3d99e3fdc13cce83ac28cc3e720399f484ae/tools/BUILD#L114)):*
> ```Python
> exports_files(
>     ["gflags.cps"],
>     visibility = ["//visibility:public"],
> ```

`visibility = ["//:__pkg__"],` should be sufficient if you can put the rest in `BUILD.bazel`.

---

*[tools/gflags.cps, line 4 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6323#-KmWbob3Vnz9RS1sDlhV:-KmWbob3Vnz9RS1sDlhW:bwmikwt) ([raw file](https://github.com/robotlocomotion/drake/blob/07ee3d99e3fdc13cce83ac28cc3e720399f484ae/tools/gflags.cps#L4)):*
> ```delphi
>   "Cps-Version": "0.8.0",
>   "Name": "gflags",
>   "Description": "The gflags package contains a C++ library that implements commandline flags processing",
> ```

`A C++ library that implements command line flag processing`

---

*[tools/gflags.cps, line 13 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6323#-KmWc-kwqKDz0edU9cbu:-KmWc-kwqKDz0edU9cbv:bkynxa9) ([raw file](https://github.com/robotlocomotion/drake/blob/07ee3d99e3fdc13cce83ac28cc3e720399f484ae/tools/gflags.cps#L13)):*
> ```delphi
>       "Location": "@prefix@/lib/libgflags.so",
>       "Includes": [
>         "@prefix@/include"
> ```

`"Includes": [ "@prefix@/include"]` would be cleaner.

---

*[tools/externals/gflags/BUILD, line 1 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6323#-KmWb3xn9MdguNGevD2F:-KmWb3xn9MdguNGevD2G:b-oe67el) ([raw file](https://github.com/robotlocomotion/drake/blob/07ee3d99e3fdc13cce83ac28cc3e720399f484ae/tools/externals/gflags/BUILD#L1)):*
> ```Python
> # -*- python -*-
> ```

Not sure this is the correct location. If you can lose the aliases to simplify things, I would put them in `BUILD.bazel` just above the jchart2d rules.

---

*[tools/externals/gflags/BUILD, line 15 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6323#-KmWbEuTaAHz7Jd9k5iJ:-KmWbEuTaAHz7Jd9k5iK:b-mbanzz) ([raw file](https://github.com/robotlocomotion/drake/blob/07ee3d99e3fdc13cce83ac28cc3e720399f484ae/tools/externals/gflags/BUILD#L15)):*
> ```Python
>     name = "COPYING.txt",
>     actual = "@com_github_gflags_gflags//:COPYING.txt",
> )
> ```

Are the aliases really necessary? We only use them once.

---

*[tools/externals/gflags/BUILD, line 19 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6323#-KmWbSyfdj7bEg1s4E0u:-KmWbSyfdj7bEg1s4E0v:b-iwovyv) ([raw file](https://github.com/robotlocomotion/drake/blob/07ee3d99e3fdc13cce83ac28cc3e720399f484ae/tools/externals/gflags/BUILD#L19)):*
> ```Python
> cmake_config(
>     package = "gflags",
> )
> ```

You can write this on one line for consistency.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6323)*
<!-- Sent from Reviewable.io -->

jamiesnape commented:
Review status: 0 of 5 files reviewed at latest revision, 8 unresolved discussions.

---

*[tools/bazel.rc, line 14 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6323#-KmWcyS0xk6kwjh0zJsj:-KmWcyS0xk6kwjh0zJsk:bsxdmle) ([raw file](https://github.com/robotlocomotion/drake/blob/07ee3d99e3fdc13cce83ac28cc3e720399f484ae/tools/bazel.rc#L14)):*
> ```
> # TODO(mwoehlke) Uncomment line above and remove line below once install.bzl
> # has been updated.
> build --output_filter='^//((?!(gflags):).)*$'
> ```

This definitely does not suppress something that we would need?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6323)*
<!-- Sent from Reviewable.io -->

jamiesnape commented:
Reviewed 5 of 5 files at r1.
Review status: all files reviewed at latest revision, 9 unresolved discussions.

---

*[tools/externals/gflags/BUILD, line 21 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6323#-KmWdCxYvrrcHUSsfoPu:-KmWdCxYvrrcHUSsfoPv:by3p0ic) ([raw file](https://github.com/robotlocomotion/drake/blob/07ee3d99e3fdc13cce83ac28cc3e720399f484ae/tools/externals/gflags/BUILD#L21)):*
> ```Python
> )
> 
> install_cmake_config(package = "gflags")  # Creates rule :install_cmake_config.
> ```

Maybe we need to tweak `install_cmake_config` to name the target something like `install_cmake_config_<package>`.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6323)*
<!-- Sent from Reviewable.io -->

jamiesnape commented:
Review status: all files reviewed at latest revision, 9 unresolved discussions.

---

*[tools/externals/gflags/BUILD, line 21 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6323#-KmWdCxYvrrcHUSsfoPu:-KmWfPi-ncCTg6S53Bvm:bfgp5zr) ([raw file](https://github.com/robotlocomotion/drake/blob/07ee3d99e3fdc13cce83ac28cc3e720399f484ae/tools/externals/gflags/BUILD#L21)):*
<details><summary><i>Previously, jamiesnape (Jamie Snape) wrote…</i></summary><blockquote>

Maybe we need to tweak `install_cmake_config` to name the target something like `install_cmake_config_<package>`.
</blockquote></details>

Or easier, would be to respect a name parameter, if it does not already,

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6323)*
<!-- Sent from Reviewable.io -->

mwoehlke-kitware commented:
Review status: all files reviewed at latest revision, 11 unresolved discussions.

---

*[tools/bazel.rc, line 14 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6323#-KmWcyS0xk6kwjh0zJsj:-KmWs2SxYrLxP_n9zg8S:b-e0tv5p) ([raw file](https://github.com/robotlocomotion/drake/blob/07ee3d99e3fdc13cce83ac28cc3e720399f484ae/tools/bazel.rc#L14)):*
<details><summary><i>Previously, jamiesnape (Jamie Snape) wrote…</i></summary><blockquote>

This definitely does not suppress something that we would need?
</blockquote></details>

Indeed; this looks like it both fails to suppress what the change is meant to make it suppress, and also breaks the existing suppression. I would guess maybe `'^//|gflags:'`?

---

*[tools/BUILD, line 111 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6323#-KmWsLFD0OLYyY8sScer:-KmWsLFD0OLYyY8sSces:bhaxdif) ([raw file](https://github.com/robotlocomotion/drake/blob/07ee3d99e3fdc13cce83ac28cc3e720399f484ae/tools/BUILD#L111)):*
> ```Python
> ])
> 
> # This file needs to be declared here to be used with "cmake_config"
> ```

BTW This comment seems inconsistent; the following `.cps`s don't have comments, but this comment is singular. It seems to me either this should be plural, to apply to the new `gflags.cps` as well as the following `.cps`s, or just dropped, as I'm not sure it's really needed.

---

*[tools/gflags.cps, line 6 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6323#-KmWsr2Ia-MAwm1-RbVe:-KmWsr2JWwosvJ1myvIb:b-2mm5om) ([raw file](https://github.com/robotlocomotion/drake/blob/07ee3d99e3fdc13cce83ac28cc3e720399f484ae/tools/gflags.cps#L6)):*
> ```delphi
>   "Description": "The gflags package contains a C++ library that implements commandline flags processing",
>   "License": "BSD-3-Clause",
>   "Version": "2.2.0",
> ```

For other packages, we've been using Python to extract the actual version information. Is there no practical way to do that for gflags? (If not, there at least should be a great big comment in `WORKSPACE` to update this file when updating gflags.)

---

*[tools/gflags.cps, line 13 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6323#-KmWc-kwqKDz0edU9cbu:-KmWslr_GRBzP2AKuTaT:byix4lx) ([raw file](https://github.com/robotlocomotion/drake/blob/07ee3d99e3fdc13cce83ac28cc3e720399f484ae/tools/gflags.cps#L13)):*
<details><summary><i>Previously, jamiesnape (Jamie Snape) wrote…</i></summary><blockquote>

`"Includes": [ "@prefix@/include"]` would be cleaner.
</blockquote></details>

Indeed; no need to break a one-item list...

---

*[tools/externals/gflags/BUILD, line 21 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6323#-KmWdCxYvrrcHUSsfoPu:-KmWtLZi8AU1wXQTRQ7X:b-o4njin) ([raw file](https://github.com/robotlocomotion/drake/blob/07ee3d99e3fdc13cce83ac28cc3e720399f484ae/tools/externals/gflags/BUILD#L21)):*
<details><summary><i>Previously, jamiesnape (Jamie Snape) wrote…</i></summary><blockquote>

Or easier, would be to respect a name parameter, if it does not already,
</blockquote></details>

I vote for adding `name = "install_cmake_config"` to the macro. The default name makes sense in the usual case that the rule is local to the package (where having the package name in the rule name also is redundant), but would allow overriding it when necessary.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6323)*
<!-- Sent from Reviewable.io -->

jwnimmer-tri commented:
*[tools/bazel.rc, line 14 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6323#-KmWcyS0xk6kwjh0zJsj:-KmX5U1kbGizm4FYuwsJ:bvkipc8) ([raw file](https://github.com/robotlocomotion/drake/blob/07ee3d99e3fdc13cce83ac28cc3e720399f484ae/tools/bazel.rc#L14)):*
<details><summary><i>Previously, mwoehlke-kitware (Matthew Woehlke) wrote…</i></summary><blockquote>

Indeed; this looks like it both fails to suppress what the change is meant to make it suppress, and also breaks the existing suppression. I would guess maybe `'^//|gflags:'`?
</blockquote></details>

FYI If `install.bzl` is anyway going to be fixed so that the warning goes away, perhaps its not worth trying to mute it?  We have ~3 warnings in the Bazel build already, a couple more for a short period of time doesn't seem like a big deal.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6323)*
<!-- Sent from Reviewable.io -->

fbudin69500 commented:
Review status: all files reviewed at latest revision, 12 unresolved discussions.

---

*[BUILD.bazel, line 42 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6323#-KmX68WDT3c7S-lAc2w5:-KmX68WF0KLi7cQOzyhP:bx21va7) ([raw file](https://github.com/robotlocomotion/drake/blob/aac87aaf1df28f79691a8fd9b8fec2ece3652553/BUILD.bazel#L42)):*
> '^//|gflags:'

---

*[tools/bazel.rc, line 12 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6323#-KmWcJTfiyvZyloFOc2a:-KmXxWMmw_scdNuwyiKF:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/07ee3d99e3fdc13cce83ac28cc3e720399f484ae/tools/bazel.rc#L12)):*
<details><summary><i>Previously, jamiesnape (Jamie Snape) wrote…</i></summary><blockquote>

TODOs should have the author of the TODO's, not the intended resolver. Here, I guess`TODO(fbudin69500, mwoehlke-kitware)` would probably be fine, though.
</blockquote></details>

Done.

---

*[tools/bazel.rc, line 14 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6323#-KmWcyS0xk6kwjh0zJsj:-KmXxXK6ZrZD0VsCUp8h:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/07ee3d99e3fdc13cce83ac28cc3e720399f484ae/tools/bazel.rc#L14)):*
<details><summary><i>Previously, jwnimmer-tri (Jeremy Nimmer) wrote…</i></summary><blockquote>

FYI If `install.bzl` is anyway going to be fixed so that the warning goes away, perhaps its not worth trying to mute it?  We have ~3 warnings in the Bazel build already, a couple more for a short period of time doesn't seem like a big deal.
</blockquote></details>

Done.

---

*[tools/BUILD, line 111 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6323#-KmWsLFD0OLYyY8sScer:-KmXyu1y_n0Ql81TX4RZ:b-wpdcth) ([raw file](https://github.com/robotlocomotion/drake/blob/07ee3d99e3fdc13cce83ac28cc3e720399f484ae/tools/BUILD#L111)):*
> inconsistent
The cps file currently needs to be located in "tools". It makes sense for the other tools that have their BUILD file also located in this folder, but it is singular for gflags that has its own bazel build system and has install rules in tools/externals/gflags

---

*[tools/BUILD, line 114 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6323#-KmWb-hT76VKs02Y1cTR:-KmXxykHErBZaC_OCGHK:b-3cbgc8) ([raw file](https://github.com/robotlocomotion/drake/blob/07ee3d99e3fdc13cce83ac28cc3e720399f484ae/tools/BUILD#L114)):*
> visibility = ["//:__pkg__"],
I get the following error if I replace the visibility by your suggestion:
ERROR: /home/fbudin/Devel/drake/tools/externals/gflags/BUILD:12:1: Target '@drake//tools:gflags.cps' is not visible from target '//tools/externals/gflags:cmake_exports'. Check the visibility declaration of the former target if you think the dependency is legitimate.

---

*[tools/gflags.cps, line 4 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6323#-KmWbob3Vnz9RS1sDlhV:-KmXzJfs_y98UIbem9-t:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/07ee3d99e3fdc13cce83ac28cc3e720399f484ae/tools/gflags.cps#L4)):*
<details><summary><i>Previously, jamiesnape (Jamie Snape) wrote…</i></summary><blockquote>

`A C++ library that implements command line flag processing`
</blockquote></details>

Done.

---

*[tools/gflags.cps, line 6 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6323#-KmWsr2Ia-MAwm1-RbVe:-KmX71rpiNO3kVxywxru:bjkosj6) ([raw file](https://github.com/robotlocomotion/drake/blob/07ee3d99e3fdc13cce83ac28cc3e720399f484ae/tools/gflags.cps#L6)):*
> s, we've been using Python to extract the a
+1 for the comment in WORKSPACE. No there is no easy way to extract the version information. The version information is available in the CMakeLists.txt gflags file which is not exposed in bazel.

---

*[tools/gflags.cps, line 13 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6323#-KmWc-kwqKDz0edU9cbu:-KmXvvl1KV-Pk7nQiCNT:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/07ee3d99e3fdc13cce83ac28cc3e720399f484ae/tools/gflags.cps#L13)):*
<details><summary><i>Previously, mwoehlke-kitware (Matthew Woehlke) wrote…</i></summary><blockquote>

Indeed; no need to break a one-item list...
</blockquote></details>

Done.

---

*[tools/externals/gflags/BUILD, line 1 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6323#-KmWb3xn9MdguNGevD2F:-KmXzNSVTnJAcgFlsSbL:b-czhrma) ([raw file](https://github.com/robotlocomotion/drake/blob/07ee3d99e3fdc13cce83ac28cc3e720399f484ae/tools/externals/gflags/BUILD#L1)):*
<details><summary><i>Previously, jamiesnape (Jamie Snape) wrote…</i></summary><blockquote>

Not sure this is the correct location. If you can lose the aliases to simplify things, I would put them in `BUILD.bazel` just above the jchart2d rules.
</blockquote></details>

The first alias for gflags is required to use cmake_config

---

*[tools/externals/gflags/BUILD, line 15 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6323#-KmWbEuTaAHz7Jd9k5iJ:-KmXwjjzPHjrucg5yie4:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/07ee3d99e3fdc13cce83ac28cc3e720399f484ae/tools/externals/gflags/BUILD#L15)):*
<details><summary><i>Previously, jamiesnape (Jamie Snape) wrote…</i></summary><blockquote>

Are the aliases really necessary? We only use them once.
</blockquote></details>

Done.

---

*[tools/externals/gflags/BUILD, line 19 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6323#-KmWbSyfdj7bEg1s4E0u:-KmXwq287Jetw7u0yJmJ:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/07ee3d99e3fdc13cce83ac28cc3e720399f484ae/tools/externals/gflags/BUILD#L19)):*
<details><summary><i>Previously, jamiesnape (Jamie Snape) wrote…</i></summary><blockquote>

You can write this on one line for consistency.
</blockquote></details>

Done.

---

*[tools/externals/gflags/BUILD, line 21 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6323#-KmWdCxYvrrcHUSsfoPu:-KmXxQyDPiPF279LEYIg:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/07ee3d99e3fdc13cce83ac28cc3e720399f484ae/tools/externals/gflags/BUILD#L21)):*
<details><summary><i>Previously, mwoehlke-kitware (Matthew Woehlke) wrote…</i></summary><blockquote>

I vote for adding `name = "install_cmake_config"` to the macro. The default name makes sense in the usual case that the rule is local to the package (where having the package name in the rule name also is redundant), but would allow overriding it when necessary.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6323)*
<!-- Sent from Reviewable.io -->

jamiesnape commented:
Review status: 0 of 7 files reviewed at latest revision, 12 unresolved discussions.

---

*[tools/BUILD, line 111 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6323#-KmWsLFD0OLYyY8sScer:-KmY--WPd0gWsbx19Kzc:boxeqce) ([raw file](https://github.com/robotlocomotion/drake/blob/07ee3d99e3fdc13cce83ac28cc3e720399f484ae/tools/BUILD#L111)):*
<details><summary><i>Previously, fbudin69500 (Francois Budin) wrote…</i></summary><blockquote>

> inconsistent
The cps file currently needs to be located in "tools". It makes sense for the other tools that have their BUILD file also located in this folder, but it is singular for gflags that has its own bazel build system and has install rules in tools/externals/gflags
</blockquote></details>

I don't think it is needed.

---

*[tools/BUILD, line 114 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6323#-KmWb-hT76VKs02Y1cTR:-KmY-5cQHhZ2O83HuQC9:ba27t0s) ([raw file](https://github.com/robotlocomotion/drake/blob/07ee3d99e3fdc13cce83ac28cc3e720399f484ae/tools/BUILD#L114)):*
<details><summary><i>Previously, fbudin69500 (Francois Budin) wrote…</i></summary><blockquote>

> visibility = ["//:__pkg__"],
I get the following error if I replace the visibility by your suggestion:
ERROR: /home/fbudin/Devel/drake/tools/externals/gflags/BUILD:12:1: Target '@drake//tools:gflags.cps' is not visible from target '//tools/externals/gflags:cmake_exports'. Check the visibility declaration of the former target if you think the dependency is legitimate.

</blockquote></details>

Since you did not move the rule to `BUILD.bazel`, then it will not work. It needs to be something different.

---

*[tools/externals/gflags/BUILD, line 1 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6323#-KmWb3xn9MdguNGevD2F:-KmY-KUJ_gTTsAA7eD4i:b-j4zg3f) ([raw file](https://github.com/robotlocomotion/drake/blob/07ee3d99e3fdc13cce83ac28cc3e720399f484ae/tools/externals/gflags/BUILD#L1)):*
<details><summary><i>Previously, fbudin69500 (Francois Budin) wrote…</i></summary><blockquote>

The first alias for gflags is required to use cmake_config
</blockquote></details>

Let's put everything in `tools/install/com_github_gflags_gflags` or `tools/install/gflags`.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6323)*
<!-- Sent from Reviewable.io -->

jamiesnape commented:
Review status: 0 of 7 files reviewed at latest revision, 10 unresolved discussions.

---

*[tools/bazel.rc, line 12 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6323#-KmWcJTfiyvZyloFOc2a:-KmY01VZvzwk1qHHcP9U:b15v83t) ([raw file](https://github.com/robotlocomotion/drake/blob/07ee3d99e3fdc13cce83ac28cc3e720399f484ae/tools/bazel.rc#L12)):*
<details><summary><i>Previously, fbudin69500 (Francois Budin) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

`mwoehlke-kitware`

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6323)*
<!-- Sent from Reviewable.io -->

mwoehlke-kitware commented:
BTW (meta), when amending commits, you might want to consider getting in the habit of usually using `--reset-author` to reset the time on the commit, which arguably makes more sense when the commit has been edited.

---

Reviewed 2 of 7 files at r2, 5 of 6 files at r3, 2 of 6 files at r5.
Review status: 3 of 11 files reviewed at latest revision, 14 unresolved discussions.

---

*[tools/BUILD, line 122 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/6323#-KmY4CTEH_5ucg7LpJdX:-KmY4CTFgeXOFCy-Bjjv:bt7gzer) ([raw file](https://github.com/robotlocomotion/drake/blob/7d846f51343504a10bdab6cc4fbe0ef2ab82f180/tools/BUILD#L122)):*
> ```Python
> exports_files(
>     ["gflags.cps"],
>     visibility = ["//visibility:public"],
> ```

Try `//tools/externals/gflags/__pkg__`? (See https://bazel.build/versions/master/docs/be/common-definitions.html#common.visibility for hints; at any rate, this can probably be more specific.)

---

*[tools/gflags.cps, line 12 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/6323#-KmY50CTogJdoDvgrvjy:-KmY50CUzs3WgShIVAHO:b-1pd3tb) ([raw file](https://github.com/robotlocomotion/drake/blob/7d846f51343504a10bdab6cc4fbe0ef2ab82f180/tools/gflags.cps#L12)):*
> ```delphi
>       "Type": "dylib",
>       "Location": "@prefix@/lib/libgflags.so",
>       "Includes": [ "@prefix@/include" ]
> ```

FYI, while we are not especially consistent, it is slightly more common to omit spaces inside `[]`s for single-item lists.

---

*[tools/install.bzl, line 307 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/6323#-KmY6nZ5BXsDazI-LZVN:-KmY6nZ5BXsDazI-LZVO:b-1lbfie) ([raw file](https://github.com/robotlocomotion/drake/blob/7d846f51343504a10bdab6cc4fbe0ef2ab82f180/tools/install.bzl#L307)):*
> ```Python
>     license_docs: List of license files to install (uses doc_dest).
>     guess_hdrs: See note.
>     guess_hdrs_exclude: Headers found with guess_hdrs to exclude from install
> ```

"List of headers found by \`\`guess_headers\`\` to exclude from installation." please :smile:. (Also, note the "." and the "\`\`"s.)

---

*[tools/install.bzl, line 457 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/6323#-KmY7ZkPP0Nbzqo1T9Ll:-KmY7ZkPP0Nbzqo1T9Lm:blxorkv) ([raw file](https://github.com/robotlocomotion/drake/blob/7d846f51343504a10bdab6cc4fbe0ef2ab82f180/tools/install.bzl#L457)):*
> ```Python
> 
> #------------------------------------------------------------------------------
> def install_cmake_config(package, versioned=True, name = "install_cmake_config"):
> ```

...buildifier didn't flag the inconsistent space around `=`? (Granted, not introduced by this PR, but since you're changing this line... which, also is too long...)

Also, while I'm inclined to keep this anyway, I note with amusement that we aren't actually using it.

---

*[tools/install/gflags/BUILD, line 7 at r5](https://reviewable.io:443/reviews/robotlocomotion/drake/6323#-KmY7ynw5iVBn1-ECIjB:-KmY7ynxoBONDbtxKbFw:bjt1ite) ([raw file](https://github.com/robotlocomotion/drake/blob/ab839e2e3a79bb200cb3687eddc7e6d0e54937d5/tools/install/gflags/BUILD#L7)):*
> ```Python
> package(default_visibility = ["//visibility:public"])
> 
> cmake_config( package = "gflags" )
> ```

Remove spaces inside `()`s...

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6323#-:-KmY5_BRvrqGebyDEktK:bnlxklo)*
<!-- Sent from Reviewable.io -->

mwoehlke-kitware commented:
Reviewed 1 of 10 files at r4, 2 of 6 files at r5.
Review status: 5 of 11 files reviewed at latest revision, 11 unresolved discussions.

---

*[tools/gflags.cps, line 6 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6323#-KmWsr2Ia-MAwm1-RbVe:-KmY9Cdjh2Dr5-623CBH:bl2zucw) ([raw file](https://github.com/robotlocomotion/drake/blob/07ee3d99e3fdc13cce83ac28cc3e720399f484ae/tools/gflags.cps#L6)):*
<details><summary><i>Previously, fbudin69500 (Francois Budin) wrote…</i></summary><blockquote>

> s, we've been using Python to extract the a
+1 for the comment in WORKSPACE. No there is no easy way to extract the version information. The version information is available in the CMakeLists.txt gflags file which is not exposed in bazel.
</blockquote></details>

Eew... although I guess that makes sense in Google's world; you don't ever need to know what version of gflags you are using, because your code is tightly coupled to gflags...

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6323)*
<!-- Sent from Reviewable.io -->

jamiesnape commented:
Review status: 5 of 11 files reviewed at latest revision, 10 unresolved discussions.

---

*[tools/BUILD, line 122 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/6323#-KmY4CTEH_5ucg7LpJdX:-KmYGpwZATnEc_1ydme7:b-hzvbdt) ([raw file](https://github.com/robotlocomotion/drake/blob/7d846f51343504a10bdab6cc4fbe0ef2ab82f180/tools/BUILD#L122)):*
<details><summary><i>Previously, mwoehlke-kitware (Matthew Woehlke) wrote…</i></summary><blockquote>

Try `//tools/externals/gflags/__pkg__`? (See https://bazel.build/versions/master/docs/be/common-definitions.html#common.visibility for hints; at any rate, this can probably be more specific.)
</blockquote></details>

`//tools/externals/gflags:__pkg__` probably, or `//tools/install/gflags:__pkg__` when you move it.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6323)*
<!-- Sent from Reviewable.io -->

fbudin69500 commented:
Review status: 1 of 10 files reviewed at latest revision, 10 unresolved discussions.

---

*[tools/install.bzl, line 307 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/6323#-KmY6nZ5BXsDazI-LZVN:-KmYauha-67D9qfkHP2N:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/7d846f51343504a10bdab6cc4fbe0ef2ab82f180/tools/install.bzl#L307)):*
<details><summary><i>Previously, mwoehlke-kitware (Matthew Woehlke) wrote…</i></summary><blockquote>

"List of headers found by \`\`guess_headers\`\` to exclude from installation." please :smile:. (Also, note the "." and the "\`\`"s.)
</blockquote></details>

Done.

---

*[tools/install.bzl, line 457 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/6323#-KmY7ZkPP0Nbzqo1T9Ll:-KmYYvVC6G0a16hLNQel:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/7d846f51343504a10bdab6cc4fbe0ef2ab82f180/tools/install.bzl#L457)):*
<details><summary><i>Previously, mwoehlke-kitware (Matthew Woehlke) wrote…</i></summary><blockquote>

...buildifier didn't flag the inconsistent space around `=`? (Granted, not introduced by this PR, but since you're changing this line... which, also is too long...)

Also, while I'm inclined to keep this anyway, I note with amusement that we aren't actually using it.
</blockquote></details>

Done.

---

*[tools/install/gflags/BUILD, line 7 at r5](https://reviewable.io:443/reviews/robotlocomotion/drake/6323#-KmY7ynw5iVBn1-ECIjB:-KmYZ8imHd56Nsl8scRD:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/ab839e2e3a79bb200cb3687eddc7e6d0e54937d5/tools/install/gflags/BUILD#L7)):*
<details><summary><i>Previously, mwoehlke-kitware (Matthew Woehlke) wrote…</i></summary><blockquote>

Remove spaces inside `()`s...
</blockquote></details>

Done.

---

*[tools/BUILD, line 111 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6323#-KmWsLFD0OLYyY8sScer:-KmY7N887Deml01aYtbB:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/07ee3d99e3fdc13cce83ac28cc3e720399f484ae/tools/BUILD#L111)):*
<details><summary><i>Previously, jamiesnape (Jamie Snape) wrote…</i></summary><blockquote>

I don't think it is needed.
</blockquote></details>

Done.

---

*[tools/BUILD, line 114 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6323#-KmWb-hT76VKs02Y1cTR:-KmYaxqYznXbWRelUaSj:b-9vs292) ([raw file](https://github.com/robotlocomotion/drake/blob/07ee3d99e3fdc13cce83ac28cc3e720399f484ae/tools/BUILD#L114)):*
<details><summary><i>Previously, jamiesnape (Jamie Snape) wrote…</i></summary><blockquote>

Since you did not move the rule to `BUILD.bazel`, then it will not work. It needs to be something different.
</blockquote></details>

See comment below (cps file moved)

---

*[tools/BUILD, line 122 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/6323#-KmY4CTEH_5ucg7LpJdX:-KmYXTUz4a9pjbkbySmX:b-3inyc9) ([raw file](https://github.com/robotlocomotion/drake/blob/7d846f51343504a10bdab6cc4fbe0ef2ab82f180/tools/BUILD#L122)):*
<details><summary><i>Previously, jamiesnape (Jamie Snape) wrote…</i></summary><blockquote>

`//tools/externals/gflags:__pkg__` probably, or `//tools/install/gflags:__pkg__` when you move it.
</blockquote></details>

I tried all that and nothing worked. I added an option to manually set the cps script name in `cmake_config` and moved `gflags.cps` in `tools/install/gflags/`

---

*[tools/gflags.cps, line 6 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6323#-KmWsr2Ia-MAwm1-RbVe:-KmYA0RIeFWA7BPo8uf4:bdg1nuw) ([raw file](https://github.com/robotlocomotion/drake/blob/07ee3d99e3fdc13cce83ac28cc3e720399f484ae/tools/gflags.cps#L6)):*
<details><summary><i>Previously, mwoehlke-kitware (Matthew Woehlke) wrote…</i></summary><blockquote>

Eew... although I guess that makes sense in Google's world; you don't ever need to know what version of gflags you are using, because your code is tightly coupled to gflags...
</blockquote></details>

I was hoping that hoping that I could use the file "config.,h" that is configured by bazel, but with bazel it does not actually configure the version number.

---

*[tools/gflags.cps, line 12 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/6323#-KmY50CTogJdoDvgrvjy:-KmYY8gjgAb_pHItJnIS:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/7d846f51343504a10bdab6cc4fbe0ef2ab82f180/tools/gflags.cps#L12)):*
<details><summary><i>Previously, mwoehlke-kitware (Matthew Woehlke) wrote…</i></summary><blockquote>

FYI, while we are not especially consistent, it is slightly more common to omit spaces inside `[]`s for single-item lists.
</blockquote></details>

Done.

---

*[tools/externals/gflags/BUILD, line 1 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6323#-KmWb3xn9MdguNGevD2F:-KmY6x2IuzPzzF6uKzGG:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/07ee3d99e3fdc13cce83ac28cc3e720399f484ae/tools/externals/gflags/BUILD#L1)):*
<details><summary><i>Previously, jamiesnape (Jamie Snape) wrote…</i></summary><blockquote>

Let's put everything in `tools/install/com_github_gflags_gflags` or `tools/install/gflags`.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6323)*
<!-- Sent from Reviewable.io -->

mwoehlke-kitware commented:
Reviewed 2 of 6 files at r6, 5 of 5 files at r7.
Review status: all files reviewed at latest revision, 11 unresolved discussions.

---

*[tools/install.bzl, line 307 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/6323#-KmY6nZ5BXsDazI-LZVN:-KmaalM5k1m2B-9OzvUS:b-fhehrl) ([raw file](https://github.com/robotlocomotion/drake/blob/7d846f51343504a10bdab6cc4fbe0ef2ab82f180/tools/install.bzl#L307)):*
<details><summary><i>Previously, fbudin69500 (Francois Budin) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Um... looks like you applied the change in the wrong spot?

---

*[tools/install.bzl, line 457 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/6323#-KmY7ZkPP0Nbzqo1T9Ll:-Kmabn49FGqsVTIu3BZj:b-5qrygh) ([raw file](https://github.com/robotlocomotion/drake/blob/7d846f51343504a10bdab6cc4fbe0ef2ab82f180/tools/install.bzl#L457)):*
<details><summary><i>Previously, fbudin69500 (Francois Budin) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

FYI Actually, Bazel convention (or at least Drake convention?) appears to be to *have* spaces here (which is contrary to Python convention, I know). That, or we've just been consistently doing it wrong.

I'm thinking let's go ahead and leave this as-is (i.e. following Python convention), and I'm strongly considering fixing it elsewhere and then *enforcing* it via `pycodestyle` (i.e. modifying `python_lint` to also check Bazel files, though without the idiotic and readability-harming E265+E302).

---

*[tools/install.bzl, line 77 at r7](https://reviewable.io:443/reviews/robotlocomotion/drake/6323#-Kmaa9Y0HE41WIY7UKP9:-Kmaa9Y2Ydi8RX_irjfj:b-hm77p3) ([raw file](https://github.com/robotlocomotion/drake/blob/4193860f675150366a85be0aef22733278dd91df/tools/install.bzl#L77)):*
> ```Python
>             that is used as the default when there is no entry for the specific
>             extension.
>         excluded_files (:obj:`list` of :obj:`str`): List of headers found by
> ```

Actually, the old text *here* was okay; at this point, `excluded_files` isn't necessarily headers. (I'll take "to exclude from" vs. "excluded from", though, but really, either is okay.)

Besides, it's `guess_hdrs`, not `guess_headers` :smile:. (D'oh! My bad, I see that was *my* mistake in the previous comment on the review :flushed:.)

---

*[tools/install.bzl, line 420 at r7](https://reviewable.io:443/reviews/robotlocomotion/drake/6323#-KmabF1u08pux11RP8sB:-KmabF1x2fPnZ-NxpT4o:bk7njg) ([raw file](https://github.com/robotlocomotion/drake/blob/4193860f675150366a85be0aef22733278dd91df/tools/install.bzl#L420)):*
> ```Python
>     if script and version_file:
>         if cps_file_name:
>             fail("`cps_file_name` should not be set if"
> ```

BTW, is there precedent for using backticks around parameter names in error messages?

---

*[tools/gflags.cps, line 6 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6323#-KmWsr2Ia-MAwm1-RbVe:-Kmaie-eQJOkCiAa0yVJ:bdizauc) ([raw file](https://github.com/robotlocomotion/drake/blob/07ee3d99e3fdc13cce83ac28cc3e720399f484ae/tools/gflags.cps#L6)):*
<details><summary><i>Previously, fbudin69500 (Francois Budin) wrote…</i></summary><blockquote>

I was hoping that hoping that I could use the file "config.,h" that is configured by bazel, but with bazel it does not actually configure the version number.
</blockquote></details>

Right, Google believes that when using Bazel, no one needs to know version numbers :stuck_out_tongue:.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6323)*
<!-- Sent from Reviewable.io -->

fbudin69500 commented:
Review status: all files reviewed at latest revision, 10 unresolved discussions.

---

*[tools/install.bzl, line 307 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/6323#-KmY6nZ5BXsDazI-LZVN:-KmanTUVUjQmO-8hzqt4:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/7d846f51343504a10bdab6cc4fbe0ef2ab82f180/tools/install.bzl#L307)):*
<details><summary><i>Previously, mwoehlke-kitware (Matthew Woehlke) wrote…</i></summary><blockquote>

Um... looks like you applied the change in the wrong spot?
</blockquote></details>

Done.

---

*[tools/install.bzl, line 457 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/6323#-KmY7ZkPP0Nbzqo1T9Ll:-KmaozaLBeKHocZ3h_KS:b-gqtjyj) ([raw file](https://github.com/robotlocomotion/drake/blob/7d846f51343504a10bdab6cc4fbe0ef2ab82f180/tools/install.bzl#L457)):*
<details><summary><i>Previously, mwoehlke-kitware (Matthew Woehlke) wrote…</i></summary><blockquote>

FYI Actually, Bazel convention (or at least Drake convention?) appears to be to *have* spaces here (which is contrary to Python convention, I know). That, or we've just been consistently doing it wrong.

I'm thinking let's go ahead and leave this as-is (i.e. following Python convention), and I'm strongly considering fixing it elsewhere and then *enforcing* it via `pycodestyle` (i.e. modifying `python_lint` to also check Bazel files, though without the idiotic and readability-harming E265+E302).
</blockquote></details>

I found both in the source code, but I will leave the space here.

---

*[tools/install.bzl, line 77 at r7](https://reviewable.io:443/reviews/robotlocomotion/drake/6323#-Kmaa9Y0HE41WIY7UKP9:-KmanU36Vk31R9cZ2nF1:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/4193860f675150366a85be0aef22733278dd91df/tools/install.bzl#L77)):*
<details><summary><i>Previously, mwoehlke-kitware (Matthew Woehlke) wrote…</i></summary><blockquote>

Actually, the old text *here* was okay; at this point, `excluded_files` isn't necessarily headers. (I'll take "to exclude from" vs. "excluded from", though, but really, either is okay.)

Besides, it's `guess_hdrs`, not `guess_headers` :smile:. (D'oh! My bad, I see that was *my* mistake in the previous comment on the review :flushed:.)
</blockquote></details>

Done.

---

*[tools/install.bzl, line 420 at r7](https://reviewable.io:443/reviews/robotlocomotion/drake/6323#-KmabF1u08pux11RP8sB:-KmaoAKQqTwlCYY-6TkP:b-5rue5w) ([raw file](https://github.com/robotlocomotion/drake/blob/4193860f675150366a85be0aef22733278dd91df/tools/install.bzl#L420)):*
<details><summary><i>Previously, mwoehlke-kitware (Matthew Woehlke) wrote…</i></summary><blockquote>

BTW, is there precedent for using backticks around parameter names in error messages?
</blockquote></details>

No. The only example I found now grepping through the source code (in tools/lcm.bzl) actually has no markups for the parameter names. Updating this.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6323)*
<!-- Sent from Reviewable.io -->

mwoehlke-kitware commented:
Reviewed 1 of 1 files at r8.
Review status: all files reviewed at latest revision, 8 unresolved discussions.

---

*[tools/install.bzl, line 457 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/6323#-KmY7ZkPP0Nbzqo1T9Ll:-Kmau60MNdGzgjyPYUKT:b-re86n5) ([raw file](https://github.com/robotlocomotion/drake/blob/7d846f51343504a10bdab6cc4fbe0ef2ab82f180/tools/install.bzl#L457)):*
<details><summary><i>Previously, fbudin69500 (Francois Budin) wrote…</i></summary><blockquote>

I found both in the source code, but I will leave the space here.
</blockquote></details>

Okay (for now / for this PR).

---

*[tools/install.bzl, line 420 at r7](https://reviewable.io:443/reviews/robotlocomotion/drake/6323#-KmabF1u08pux11RP8sB:-KmatxROSaY00DoZHqG2:b-ucfkn4) ([raw file](https://github.com/robotlocomotion/drake/blob/4193860f675150366a85be0aef22733278dd91df/tools/install.bzl#L420)):*
<details><summary><i>Previously, fbudin69500 (Francois Budin) wrote…</i></summary><blockquote>

No. The only example I found now grepping through the source code (in tools/lcm.bzl) actually has no markups for the parameter names. Updating this.
</blockquote></details>

Thanks.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6323)*
<!-- Sent from Reviewable.io -->

mwoehlke-kitware commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: all files reviewed at latest revision, 9 unresolved discussions.

---

*[WORKSPACE, line 55 at r8](https://reviewable.io:443/reviews/robotlocomotion/drake/6323#-Kmb-u4uVzvJIiqJUF7H:-Kmb-u4x9Y8xDHz-WMOA:bfj62vn) ([raw file](https://github.com/robotlocomotion/drake/blob/9421e8fdc5863bca4c4368de668d246e6d2dd93f/WORKSPACE#L55)):*
> ```Python
> 
> # When updating the version of gflags, update tools/gflags.cps
> # TODO(fbudin69500): Create tools/gflags.cps from a configuration file
> ```

BTW, based on other comments, is this even possible? If not, I suppose the comment is not useful. (If yes, definitely keep it.)

---

*[tools/install.bzl, line 89 at r8](https://reviewable.io:443/reviews/robotlocomotion/drake/6323#-Kmb0X0bCMZ_CsrWU0gB:-Kmb0X0ep2J2uMMWjn9s:b-st1sdl) ([raw file](https://github.com/robotlocomotion/drake/blob/9421e8fdc5863bca4c4368de668d246e6d2dd93f/tools/install.bzl#L89)):*
> ```Python
>     for f in file_labels:
>         for a in f.files:
>             if _output_path(ctx,a,[]) in excluded_files:
> ```

BTW, missing spaces after `,`s?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6323#-:-Kmb1QcFlseZZuhiLCPi:bnfp4nl)*
<!-- Sent from Reviewable.io -->

jamiesnape commented:
@drake-jenkins-bot linux-gcc-experimental-matlab please

jamiesnape commented:
Reviewed 2 of 7 files at r2, 1 of 10 files at r4, 2 of 6 files at r6, 4 of 5 files at r7, 1 of 1 files at r8.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[WORKSPACE, line 55 at r8](https://reviewable.io:443/reviews/robotlocomotion/drake/6323#-Kmb-u4uVzvJIiqJUF7H:-KmbCRuKnmQ8pe_laAH3:b-i6qqf9) ([raw file](https://github.com/robotlocomotion/drake/blob/9421e8fdc5863bca4c4368de668d246e6d2dd93f/WORKSPACE#L55)):*
<details><summary><i>Previously, mwoehlke-kitware (Matthew Woehlke) wrote…</i></summary><blockquote>

BTW, based on other comments, is this even possible? If not, I suppose the comment is not useful. (If yes, definitely keep it.)
</blockquote></details>

I don't think it is possible. Remove the TODO.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6323)*
<!-- Sent from Reviewable.io -->

jamiesnape commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: all files reviewed at latest revision, 3 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6323#-:-KmbD8uIuIo_zhv-OpQT:bnfp4nl)*
<!-- Sent from Reviewable.io -->

jamiesnape commented:
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[WORKSPACE, line 55 at r8](https://reviewable.io:443/reviews/robotlocomotion/drake/6323#-Kmb-u4uVzvJIiqJUF7H:-KmbE5CxTNuyUvU2r7zE:buxz9zs) ([raw file](https://github.com/robotlocomotion/drake/blob/9421e8fdc5863bca4c4368de668d246e6d2dd93f/WORKSPACE#L55)):*
<details><summary><i>Previously, jamiesnape (Jamie Snape) wrote…</i></summary><blockquote>

I don't think it is possible. Remove the TODO.
</blockquote></details>

Also, you moved `gflags.cps`.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6323)*
<!-- Sent from Reviewable.io -->

fbudin69500 commented:
Review status: 4 of 6 files reviewed at latest revision, 2 unresolved discussions.

---

*[WORKSPACE, line 55 at r8](https://reviewable.io:443/reviews/robotlocomotion/drake/6323#-Kmb-u4uVzvJIiqJUF7H:-KmbmFP9_eF0ZYRMrpUw:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/9421e8fdc5863bca4c4368de668d246e6d2dd93f/WORKSPACE#L55)):*
<details><summary><i>Previously, jamiesnape (Jamie Snape) wrote…</i></summary><blockquote>

Also, you moved `gflags.cps`.
</blockquote></details>

Done.

---

*[tools/install.bzl, line 89 at r8](https://reviewable.io:443/reviews/robotlocomotion/drake/6323#-Kmb0X0bCMZ_CsrWU0gB:-KmbmVJCUBtFi5OQtuhI:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/9421e8fdc5863bca4c4368de668d246e6d2dd93f/tools/install.bzl#L89)):*
<details><summary><i>Previously, mwoehlke-kitware (Matthew Woehlke) wrote…</i></summary><blockquote>

BTW, missing spaces after `,`s?
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6323)*
<!-- Sent from Reviewable.io -->

fbudin69500 commented:
+@ggould-tri plateform review please

---

Review status: 4 of 6 files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6323#-:-KmbotyLVh4INLKXeocg:b3k0srm)*
<!-- Sent from Reviewable.io -->

ggould-tri commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 1 of 7 files at r2, 3 of 5 files at r7, 2 of 2 files at r9.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6323#-:-Kmc6Ev1iLl1ey77YMQO:bnfp4nl)*
<!-- Sent from Reviewable.io -->

jamiesnape commented:
Reviewed 2 of 2 files at r9.
Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6323)*
<!-- Sent from Reviewable.io -->
